### PR TITLE
add pure filter test for folder pipe

### PR DESCRIPTION
**Pull request by heronakam**

### CODE REVIEW

**Comment by heronakam** [comment]
@deepu105 @jdubois  Shall I continue working on this project with @g-boy05 or shall we switch to the generator-jhipster?

File: tests/language/language.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import {JhiLanguageService} from '../../src/language/language.service';
```
Comment (deepu105) [inline tests/language/language.service.spec.ts]: spacing

File: tests/language/language.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import { TestBed, inject } from '@angular/core/testing';
+import { TranslateService } from 'ng2-translate';
+import { Observable } from 'rxjs/Observable';
+
+class TranslateLoaderMock {
+
+    public setLocations(locaions: string[]) {
```
Comment (deepu105) [inline tests/language/language.service.spec.ts]: end `{}` in the same line as there is no method body

File: tests/language/language.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                    {
+                        provide: TranslateService,
+                        useClass: TranslateServiceMock
+                    }
+                ]
+            });
+        });
+
+        it('should changeLanguage', inject([JhiLanguageService], (service: JhiLanguageService) => {
+            service.changeLanguage('en');
```
Comment (deepu105) [inline tests/language/language.service.spec.ts]: default is `en` hence please check another language as well

File: tests/pipe/pure-filer.pipe.spec.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import {PureFilterPipe} from '../../src/pipe/pure-filter.pipe';
```
Comment (deepu105) [inline tests/pipe/pure-filer.pipe.spec.ts]: spacing

File: tests/pipe/truncateWords.pipe.spec.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    beforeEach(() => {
+        pipe = new TruncateWordsPipe();
+    });
+
+    it('Should return an empty string', () => {
+        let words: any = -1;
+        let result = pipe.transform(input, words);
+        expect(result).toBe('');
+    });
+        it('Should return sentence', () => {
```
Comment (deepu105) [inline tests/pipe/truncateWords.pipe.spec.ts]: indent

File: tests/pipe/truncateWords.pipe.spec.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import {TruncateWordsPipe} from '../../src/pipe/truncate-words.pipe';
```
Comment (deepu105) [inline tests/pipe/truncateWords.pipe.spec.ts]: spacing

File: tests/pipe/truncateWords.pipe.spec.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+describe('TruncateWordsPipe Tests', () => {
+    let pipe: TruncateWordsPipe;
+    let input: any = 'Jhipster is the best framework';
```
Comment (deepu105) [inline tests/pipe/truncateWords.pipe.spec.ts]: type is string and inferred by RHS hence remove type `any`

File: tests/pipe/truncateWords.pipe.spec.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+        pipe = new TruncateWordsPipe();
+    });
+
+    it('Should return an empty string', () => {
+        let words: any = -1;
+        let result = pipe.transform(input, words);
+        expect(result).toBe('');
+    });
+        it('Should return sentence', () => {
+        let words: any = 3;
```
Comment (deepu105) [inline tests/pipe/truncateWords.pipe.spec.ts]: remove `any`

File: tests/pipe/truncateWords.pipe.spec.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+describe('TruncateWordsPipe Tests', () => {
+    let pipe: TruncateWordsPipe;
+    let input: any = 'Jhipster is the best framework';
+    beforeEach(() => {
+        pipe = new TruncateWordsPipe();
+    });
+
+    it('Should return an empty string', () => {
+        let words: any = -1;
```
Comment (deepu105) [inline tests/pipe/truncateWords.pipe.spec.ts]: return any

**Comment by heronakam** [comment]
the all files are tested locally 
yarntest

File: tests/language/language.service.spec.ts [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+    public resetLang(lang: string) {}
+
+    public use(lang: string): Observable<any> {
+        return Observable.of();
+    }
+}
+
+
+    describe('LanguageService Test', () => {
```
Comment (deepu105) [inline tests/language/language.service.spec.ts]: why is this block indented out?

File: tests/language/language.service.spec.ts [hunk 11]
```diff
@@ -0,0 +1,53 @@
+import { JhiLanguageService } from '../../src/language/language.service';
+import { TestBed, inject } from '@angular/core/testing';
+import { TranslateService } from 'ng2-translate';
+import { Observable } from 'rxjs/Observable';
+
+class TranslateLoaderMock {
+
```
Comment (deepu105) [inline tests/language/language.service.spec.ts]: extra line. remove

File: tests/language/language.service.spec.ts [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    public setLocations(locations: string[]) {}
+
+    public resetLang(lang: string) {}
+
+    public use(lang: string): Observable<any> {
+        return Observable.of();
+    }
+}
+
+
```
Comment (deepu105) [inline tests/language/language.service.spec.ts]: extra line remove

File: tests/pipe/pure-filer.pipe.spec.ts [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+    let pipe: PureFilterPipe;
+    let list: Array<any>;
+    let filter;
+
+    beforeEach(() => {
+        pipe = new PureFilterPipe();
+
+    });
+
+it('Should filter by  string', () => {
```
Comment (deepu105) [inline tests/pipe/pure-filer.pipe.spec.ts]: indentation is wrong. Need to check why this doesnt fail lint

File: tests/pipe/pure-filer.pipe.spec.ts [hunk 14]
```diff
@@ -0,0 +1,32 @@
+import { PureFilterPipe } from '../../src/pipe/pure-filter.pipe';
+
```
Comment (deepu105) [inline tests/pipe/pure-filer.pipe.spec.ts]: extra line

File: tests/pipe/pure-filer.pipe.spec.ts [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+    let filter;
+    beforeEach(() => {
+        pipe = new PureFilterPipe();
+    });
+
+    it('Should filter by  string', () => {
+        list = ['java', 'javaScript', 'TypeScript'];
+        const result = pipe.transform(list, 'java', null);
+        expect(result).toEqual(['java', 'javaScript']);
+        });
```
Comment (deepu105) [inline tests/pipe/pure-filer.pipe.spec.ts]: wrong indent. Can you please use some plugin for your editor or use an IDE to indent the files automatically. What editor are you using?

Comment (g-boy05) [inline tests/pipe/pure-filer.pipe.spec.ts]: We use visual studio code

Comment (deepu105) [inline tests/pipe/pure-filer.pipe.spec.ts]: Right click and you will see format document. You could also install the beautify plugin if you want

**Pull request merged**